### PR TITLE
Neural Network Extension - add CMake support for HIP GPU backend

### PR DESCRIPTION
**Pull request by asalmanp**
- This PR also adds initial HIP kernel support for the gather layer.
- The support for executing the gather layer with HIP GPU backend will be added in the next PR.

### CODE REVIEW

**Comment by kiritigowda** [comment]
@asalmanp the #569 merge has a few conflicts with this PR, can you please resolve them. Thanks!

File: amd_openvx_extensions/CMakeLists.txt [hunk 1]
```diff
@@ -27,12 +27,12 @@ find_package(FFmpeg QUIET)
 find_package(AMDRPP QUIET)
 
 if(GPU_SUPPORT)
+    find_package(miopengemm PATHS ${ROCM_PATH} QUIET)
```
Comment (rrawther) [inline amd_openvx_extensions/CMakeLists.txt:30]: make sure to modify the setup to install the MIOpen for the right backend

File: amd_openvx_extensions/CMakeLists.txt [hunk 2]
```diff
@@ -85,13 +85,18 @@ if(GPU_SUPPORT)
...
 if (NEURAL_NET)
-    if(GPU_SUPPORT AND OpenCL_FOUND AND BUILD_OPENCL AND miopengemm_FOUND AND miopen_FOUND)
-        add_subdirectory(amd_nn)
-        message("-- ${Green}AMD OpenVX Neural Network Extension -- amd_nn module added${ColourReset}")
-    elseif("${BACKEND}" STREQUAL "HIP" OR "${BACKEND}" STREQUAL "CPU")
-        message("-- ${Yellow}AMD OpenVX Neural Network Extension -- amd_nn module excluded with HIP/CPU Backend${ColourReset}")
+    if(GPU_SUPPORT AND miopengemm_FOUND AND miopen_FOUND)
+        if (OpenCL_FOUND AND BUILD_OPENCL)
+            add_subdirectory(amd_nn)
+            message("-- ${Green}AMD OpenVX Neural Network Extension -- amd_nn module added${ColourReset}")
```
Comment (rrawther) [inline amd_openvx_extensions/CMakeLists.txt:91]: indicate the backend(HIP/OpenCL) in the message

Comment (asalmanp) [reply amd_openvx_extensions/CMakeLists.txt]: done.

File: amd_openvx_extensions/CMakeLists.txt (hunk 1 above)
Comment (asalmanp) [reply amd_openvx_extensions/CMakeLists.txt:30]: Yes, issue #572 is for adding the support for different backend into the setup script. The setup script changes will be added in a separate PR.

**Review by rrawther — APPROVED** [review]

**Review by rrawther — APPROVED** [review]

**Review by kiritigowda — APPROVED** [review]

**Comment by kiritigowda** [comment]
@asalmanp & @rrawther will merge this once it passes the CI

**Pull request merged**